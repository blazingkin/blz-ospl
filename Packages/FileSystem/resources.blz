:file_resource(name)
    BLZINTERNALBLZGETFILEATPATH name
end

:http_resource(name)
    name = "http://" + name
    BLZINTERNALCREATERESOURCE name
end

:https_resource(name)
    name = "https://" + name
    BLZINTERNALCREATERESOURCE name
end

:resource_from_url(name)
    BLZINTERNALCREATERESOURCE name
end

:open_resource(resource)
    BLZINTERNALOPENRESOURCE resource
end

:open(filename)
    open_resource(file_resource(filename))
end

:close(scanner)
    close_scanner(scanner)
end