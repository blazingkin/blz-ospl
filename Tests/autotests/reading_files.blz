import FileSystem

:main
	# This is kinda cheating, don't you think
	open("reading_files.out").read_lines().each(print)
end