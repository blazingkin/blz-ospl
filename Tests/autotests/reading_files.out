This
is
kinda
cheating
isn't
it?
